flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -659,14 +661,36 @@ private CompletableFuture<CleanupJobState> handleJobManagerRunnerResult(
         return jobReachedTerminalState(jobManagerRunnerResult.getExecutionGraphInfo());
     }
 
-    enum CleanupJobState {
-        LOCAL,
-        GLOBAL
+    private static class CleanupJobStateWithJobStatusIfGlobal {
+
+        private static final CleanupJobStateWithJobStatusIfGlobal LOCAL =
+                new CleanupJobStateWithJobStatusIfGlobal(null);
+
+        @Nullable private final JobStatus jobStatus;
+
+        public static CleanupJobStateWithJobStatusIfGlobal globalCleanup(JobStatus jobStatus) {
```

Comment:
strange that jobStatus isn't nullable here.

Reply:
hm, you're right. the not-null check should be moved into this method to make it fail early.